filters/include/pcl/filters/functor_filter.h

```diff
@@ -76,22 +77,40 @@ class FunctorFilter : public FilterIndices<PointT> {
    */
   void
   applyFilter(Indices& indices) override
+  {
+    applyFilter(executor::best_fit{}, indices);
+  }
+
+  template <typename Executor, typename std::enable_if_t<executor::is_instance_of_base_v<executor::inline_executor, Executor> || executor::is_instance_of_base_v<executor::omp_executor, Executor>, int> = 0>
+  void
+  applyFilter (Executor &&exec, Indices& indices)
   {
     indices.clear();
-    indices.reserve(input_->size());
+    indices.reserve(indices_->size());
+    std::vector<int> keep(indices_->size(), 0);
+    std::vector<int> remove(indices_->size(), 0);
+
     if (extract_removed_indices_) {
       removed_indices_->clear();
-      removed_indices_->reserve(input_->size());
+      removed_indices_->reserve(indices_->size());
     }
 
-    for (const auto index : *indices_) {
-      // functor returns true for points that should be selected
-      if (negative_ != functor_(*input_, index)) {
-        indices.push_back(index);
-      }
-      else if (extract_removed_indices_) {
-        removed_indices_->push_back(index);
-      }
+    auto cond_filtering = [&](index_t idx) {
+      exec.execute([&]() {
+        if (negative_ != functor_(*input_, idx)) {
+          keep[idx] = true;
+        }
+        else if (extract_removed_indices_) {
+          remove[idx] = true;
+        }
+      });
+    };
+
+    exec.bulk_execute(cond_filtering, indices_->size());
```

Comment:
num_threads of `indices->size()` might be counterproductive

Reply:
I didn't the appropriate comment, sorry about that.

So what I understood from the proposal that `bulk_execute` takes invocation count and not the number of threads.

For number of threads we would have to pass that maybe in the constuctor.
That part is yet to be done, I will update the executor design with that